.github/workflows/build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+    runs-on: windows-latest
+    env:
+      SolutionName: MouseUnSnag.sln
+      TestProjectPath: MouseUnSnagTests\MouseUnSnagTests.csproj
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Build Solution
+      shell: cmd
+      run: |
+        echo ::stop-commands::build-${{ github.run_id }}-${{ github.run_attempt }}
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```

Comment:
I don't have any experience with GitHub Actions, but I do a little with Azure Pipelines. They look similar enough that I can try to review for you ;)

Question about this path: Is it guaranteed to exist on whichever machine hosts the build action? Is there no install/setup step required?

Reply:
GitHub actions are run using environments defined at https://github.com/actions/virtual-environments, in this case "windows-latest". You bring up a good point though, it's proably better to windows-\<year\>, and in fact, this path may not exist on windows-latest anymore, since it used to point to windows-2019, but now it points to windows-2022, which has a newer version of visual studio.